common/locks/priority_semaphore_test.go

```diff
@@ -227,3 +225,24 @@ func (s *prioritySemaphoreSuite) Test_AcquireMoreThanAvailable() {
 	defer cancel()
 	s.ErrorIs(semaphore.Acquire(ctx, PriorityHigh, 2), ctx.Err())
 }
+
+// Checks if n number of threads are blocked in semaphore.
+func waitUntilBlockedInSemaphore(n int) {
+	for {
+		buf := make([]byte, 10000)
```

Comment:
I like this much better but there are two things you can improve here: 
1. You should reuse this across each iteration of the loop. Clear it with `buf = buf[:0]`
2. Sleep 100ms or so between loops. Or 50, but something

Reply:
I have added sleep and a timeout using Eventually() here.
For some reason, stack is not getting populated to buffer after resetting using `buf = buf[:0]` in the loop. So I kept it as earlier. I am curious to find the reason for this.

Reply:
I think I know why. `buf = buf[:0]` resets the slice's length (but not its capacity), and I bet it only writes up to the slice's length (not its capacity).

This code is actually incorrect: `runtime.Stack` isn't guaranteed to fill the buffer. It returns the number of bytes written.

The correct usage should look like:
```go
buf := make([]byte, 10000)
for {
	n := runtime.Stack(buf, true) // Get all goroutines' stack trace

	// Look for the specific stack trace pattern indicating blocking in targetFunction on a channel
	goroutines := bytes.Split(buf[:n], []byte("\n\n"))
	// ...
}
```

Reply:
You may also be able rewrite this whole loop using a regular expression and `FindAllSubmatchIndex`, but I'd need to see an actual stack dump to tell you

Reply:
Got it! Thanks.
Changed it to use regex.